cmd/k8s-operator/proxygroup_specs.go

```diff
@@ -167,6 +179,25 @@ func pgStatefulSet(pg *tsapi.ProxyGroup, namespace, image, tsFirewallMode string
 		return append(c.Env, envs...)
 	}()
 
+	// The pre-stop hook is used to ensure that a replica does not get terminated while cluster traffic for egress
+	// services is still being routed to it.
+	//
+	// This mechanism currently (2025-01-26) rely on the local health check being accessible on the Pod's
+	// IP, so they are not supported for ProxyGroups where users have configured TS_LOCAL_ADDR_PORT to a custom
+	// value.
```

Comment:
Not in these failover PRs, but at some point, I would like to split the egress reconcilers into a library  package. Maybe then I can also add a docs comment as to what the different reconcile sequences are and in what order the different reconcilers are expected to perform actions